lib/Makefile

```diff
@@ -234,6 +246,28 @@ lib : libzstd.a libzstd
 	@echo release build completed
 
 
+# Generate .h dependencies automatically
+
+DEPFLAGS = -MT $@ -MMD -MP -MF
+
+$(ZSTD_DYNLIB_DIR)/%.o : %.c $(ZSTD_DYNLIB_DIR)/%.d | $(ZSTD_DYNLIB_DIR)
+	@echo $@
+	$(Q)$(COMPILE.c) $(DEPFLAGS) $(ZSTD_DYNLIB_DIR)/$*.d $(OUTPUT_OPTION) $<
```

Comment:
Where is `$(COMPILE.c)` defined? It was defined in `programs/Makefile`, but not here.

Reply:
`$(COMPILE.c)` is a default variable of gnu `make`